### PR TITLE
Enums 

### PR DESCRIPTION
**Pull request by mspaulik**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -246,6 +289,12 @@ function createRdfEmitter(program: Program) {
                   writer.list(arr) as any // error in n3 typing, this is supported
                 )
               );
+            } else if (prop.type.kind === "Enum") {
```
Comment (bterlson) [inline src/index.ts]: Hopefully here all you need to do is to add a quad to say its type is an enum?

**Review by bterlson — APPROVED** [review]
This seems good to me! We can iterate from here if you'd like to merge.

**Comment by mspaulik** [comment]
@bterlson Pii custom decorator still does not work I believe. Could you have a look into that and I am happy to merge then?

**Comment by bterlson** [comment]
@mspaulik I updated everything to use the latest cadl. I'm not sure what version you're using over there, but what's here should work with 0.37.x and above. I don't think I regressed any functionality, can you take a look and see? You will need to `npm install` after you pull the changes.

There are I think two notable changes in the new versions of Cadl since this package was authored:

1. `extern dec` declarations, see `rdf.cadl` for an example. This gives you nice intellisense and makes validation easier:
image
2. Introduction of `scalar` - previously, types like integers were just models. Now they are scalars, which actually more closely aligns with how RDF works and so I think this was overall a simplification.

There were a few other modifications I made to remove usage of deprecated APIs. Anyway, try it out and let me know what you think.

**Comment by mspaulik** [comment]
Hey @bterlson 

I did look over - everything makes sense. Thanks!

Just maybe one thing until we merge this PR: I seem to be unable to specify @ispii decorator without providing value (true or false). I remember you saying that once this line is written like that  _context.call($extension, target, "x-ms-pii", **isPii ?? true);**_ , we should be able to use decorator without passing any values. 

However, I am still getting this issue : Expected 2 arguments, but got 1.Cadl(invalid-argument-count) for this model: 
@isPii
model Person {
  name: string;
  color: Colors;
}

**Comment by mspaulik** [comment]
Hey @bterlson ,

One more question regarding cadl-project.yaml. How do you specify the output folder?

Because if I dont have "cadl-output" folder manually created before running emitter, I am getting this error:
_Unhandled promise rejection!
Internal compiler error!
File issue at https://github.com/microsoft/cadl

[Error: ENOENT: no such file or directory, open 'C:\Users\Administrator\Desktop\cadl-rdf\cadl-rdfs\cadl-output\models.ttl'] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'C:\\Users\\Administrator\\Desktop\\cadl-rdf\\cadl-rdfs\\cadl-output\\models.ttl'
}_

However, if I create  "cadl-output" folder - the emitter puts the result successfully in there.

So, I guess, we need some extra line in cadl-project.yaml to handle this? Thanks!

**Comment by bterlson** [comment]
You can use the `emitFile` function from the compiler, and it will handle creating the parent directories for you. It looks something like:

```
import  { emitFile } from "@cadl-lang/compiler";
await emitFile(program, { path: "cadl-output/whatever.rdf", content: "...." });
```

**Comment by bterlson** [comment]
Hmm, I can't repro the issue you point out with `@isPii`. If I check out this branch and build, the `sample.cadl` file emits the expected output... Can you give me more precise repro steps or maybe double-check that you were checking against the latest code in this branch? Might try starting from a clean repo also, in case there are stale things sitting around.

**Comment by mspaulik** [comment]
Hey @bterlson ,

I ran again isPii decorator - all works fine now, I probably did not compile index.ts correctly before. We should be able to merge now?

**Pull request merged**